currency_old_rate_notify/models/res_currency_rate.py

```diff
@@ -0,0 +1,130 @@
+
+import logging
+from collections import defaultdict
+from datetime import timedelta
+
+from odoo import _, api, fields, models
+
+logger = logging.getLogger(__name__)
+
+MAX_DAYS_DEFAULT = 3
+
+
+class ResCurrencyRate(models.Model):
+    _inherit = "res.currency.rate"
+
+    @api.model
+    def _prepare_notify_rates_too_old(
+        self, user, cur2companies, company2name, max_days
+    ):
+        res = False
+        msg = []
+        user_company_ids = user.company_ids.ids
+        for cur_code, company_ids in cur2companies.items():
+            intersection_company_ids = list(set(company_ids) & set(user_company_ids))
+            if intersection_company_ids:
+                # Write company name in message only when there are several companies
+                if len(user_company_ids) == 1:
+                    msg.append("<li>%s</li>" % cur_code)
+                else:
+                    msg.append(
+                        "<li>%s (%s)</li>"
+                        % (
+                            cur_code,
+                            ", ".join(
+                                [
+                                    company2name[c_id]
+                                    for c_id in intersection_company_ids
+                                ]
+                            ),
+                        )
+                    )
+        if msg:
+            res = {
+                "title": _("Currency rates older than %d days") % max_days,
+                "sticky": True,
+                "message": "<ul>%s</ul>" % "".join(msg),
+            }
+        return res
+
+    @api.model
+    def _notify_rates_too_old_groups(self):
+        return self.env.ref("account.group_account_manager")
+
+    @api.model
+    def _notify_rates_get_max_days(self):
+        max_days_str = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param(
+                "currency_old_rate_notify.max_days", default=str(MAX_DAYS_DEFAULT)
+            )
+        )
+        try:
+            max_days = int(max_days_str)
+            logger.debug("currency_old_rate_warn.max_days = %s", max_days)
+        except Exception as e:
+            logger.warning(
+                "Failed to convert ir.config_parameter "
+                "currency_old_rate_warn.max_days value %s to integer. "
+                "Error: %s. Using default value %s.",
+                max_days_str,
+                e,
+                MAX_DAYS_DEFAULT,
+            )
+            max_days = MAX_DAYS_DEFAULT
+        return max_days
+
+    @api.model
+    def notify_rates_too_old(self):
+        max_days = self._notify_rates_get_max_days()
+        today = fields.Date.context_today(self)
+        limit_date = today - timedelta(max_days)
+        logger.debug("notify_rates_too_old limit_date=%s", limit_date)
+        companies = self.env["res.company"].search([])
+        currencies = self.env["res.currency"].search([])
```

Comment:
As it was stated, I agree that needs to filter only active currencies.

Reply:
that's exactly what the code does :)